pkg/secretless/plugin/sharedobj/available_plugins.go

```diff
@@ -60,35 +65,44 @@ func AllAvailablePluginsWithOptions(
 	}
 
 	httpPlugins := map[string]http.Plugin{}
+	tcpPlugins := map[string]tcp.Plugin{}
 
-	for name, httpPlugin := range internalPlugins.HTTPPlugins() {
-		if _, ok := httpPlugins[name]; ok {
-			// TODO: Should this ever happen?  Do we need this check?  Should it panic?
-			logger.Warnf("Internal plugin '%s' replaced by internal plugin", name)
+	checkHTTPNameConflict := func(pluginType, pluginName string) {
+		if _, ok := httpPlugins[pluginName]; ok {
+			logger.Panicf("%s plugin name '%s' %s", pluginType, pluginName, httpNameConflict)
```

Comment:
I'd like to hear opinions on whether this panic message is precise enough for indicating whether the plugin being checked conflicts with an `internal` vs. `external` plugin. For example, the current panic message might look like this:
```
external TCP plugin name 'FooTCP' conflicts with same-named HTTP plugin
```
It doesn't specify whether the HTTP plugin it conflicts with is internal or external. I can add this preciseness in, but it would take a little bit more bookkeeping in this function to pull it off.

Reply:
this seems good enough to ship for me!